packages/devtools_app/lib/src/memory/memory_controller.dart

```diff
@@ -459,7 +486,94 @@ class MemoryTimeline {
   }
 
   void addSample(HeapSample sample) {
+    // Always record the heap sample in the raw set of data (liveFeed).
     data.add(sample);
-    _sampleAddedNotifier.value = sample;
+
+    // Only notify that new sample has arrived if the
+    // memory source is 'Live Feed'.
+    if (!controller.offline) {
+      _sampleAddedNotifier.value = sample;
+    }
+  }
+}
+
+class MemoryLog {
+  MemoryLog(this.controller);
+
+  MemoryController controller;
+
+  /// Persist the the live memory data to a JSON file in the /tmp directory.
+  void exportMemory() {
```

Comment:
Can the export action download to Downloads in addition to/instead of tmp? A user should be able to download a file, send it to someone else, and that other person should be able to import the file. This seems like it wouldn't be possible (or would be difficult for the user to find) with downloading to tmp.

drag and drop import will be needed for other pages as well (timeline, performance, etc.)

Reply:
Yes I agree.  However, there is a limitation right now under Flutter Desktop and Flutter Web path_provider it doesn't work yet .  I'm using dart:io with package path to create a file in the tmp directory.  My concern was supporting something that works on all platforms until Flutter Desktop/Web work properly.  There's a TODO that talks about this.

Reply:
Also, drag/drop outside from OS to Flutter Desktop/Web doesn't yet exist.  Drag/drop inside flutter works fine.